edx_django_utils/cache/middleware.py

```diff
@@ -3,8 +3,6 @@
 """
 from django.utils.deprecation import MiddlewareMixin
 
-from edx_django_utils.private_utils import _check_middleware_dependencies
```

Comment:
**Reviewer Note:** I apologize that I didn't remove `_check_middleware_dependencies` first in a separate PR, but I did it after refactor, and merge conflicts would have been painful.  It should be pretty obvious and easy to review.
Please mark as resolved once you've read this.